tests/test_core.py

```diff
@@ -6245,12 +6245,12 @@ def test_dyncall_specific(self):
       self.set_setting('EXTRA_EXPORTED_RUNTIME_METHODS', exported_runtime_methods)
       self.do_run_in_out_file_test('tests', 'core', 'dyncall_specific.c')
 
-      self.set_setting('EXTRA_EXPORTED_RUNTIME_METHODS', [])
-      self.emcc_args = emcc_args + ['-s', 'DYNCALLS=1', '--js-library', path_from_root('tests', 'core', 'test_dyncalls.js')]
-      self.do_run_in_out_file_test('tests', 'core', 'test_dyncalls.c')
+    self.set_setting('EXTRA_EXPORTED_RUNTIME_METHODS', [])
+    self.emcc_args = emcc_args + ['-s', 'DYNCALLS=1', '--js-library', path_from_root('tests', 'core', 'test_dyncalls.js')]
+    self.do_run_in_out_file_test('tests', 'core', 'test_dyncalls.c')
 
-      self.emcc_args += ['-s', 'MINIMAL_RUNTIME=1', '-s', 'DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=["$dynCall"]']
-      self.do_run_in_out_file_test('tests', 'core', 'test_dyncalls.c')
+    self.emcc_args += ['-s', 'MINIMAL_RUNTIME=1', '-s', 'DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=["$dynCall"]']
+    self.do_run_in_out_file_test('tests', 'core', 'test_dyncalls.c')
```

Comment:
If these are not to be part of the loop, then they could be new tests, then - just adding `def test_NAME` etc. - ?

Reply:
They **were** a new test, until in https://github.com/emscripten-core/emscripten/pull/13296#pullrequestreview-576749490 you asked to merge them to this test?

I don't mind either way, just pick one and stick with it :)

Reply:
Sorry for not being clear!

You added a new test file, `test_dyncalls.c`. My idea was instead to add into the existing `dyncall_specific.c`. As I said there, that seems shorter, and better? An incremental addition to the existing testing, basically.

And it would be good to test the things in the loop in the addition, as all the modes matter.

If however there is a reason not to do that, and a completely new test is the way to go, then it should be in a new function(s).

Reply:
Got it, now I understand. Sorry for not realizing that!

Updated the test now, how does that look?